pkg/provenance/provenance.go

```diff
@@ -313,16 +628,17 @@ func parse() {
 		}
 
 		requestobj := event.RequestObject
+		timestamp := fmt.Sprint(event.RequestReceivedTimestamp.Format("2006-01-02 15:04:05"))
 		//now parse the spec into this provenanceObject that we found or created
-		ParseRequestObject(provObjPtr, requestobj.Raw)
+		ParseRequestObject(provObjPtr, requestobj.Raw, timestamp)
 	}
 	if err := scanner.Err(); err != nil {
 		panic(err)
 	}
 	fmt.Println("Done parsing.")
 }
 
-func ParseRequestObject(objectProvenance *ProvenanceOfObject, requestObjBytes []byte) {
+func ParseRequestObject(objectProvenance *ProvenanceOfObject, requestObjBytes []byte, timestamp string) {
```

Comment:
Not able to follow the logic for this method.

What are l1, l2, l3?

Also, did not follow the TODO comment.

Can you add an example in the comment showing the different cases that are being handled.

Reply:
this method is only to parse the requestObject bytes from the Event! So the l1,l2,l3 are just me stepping through the map structure of the requestObj, where the spec lives . 
I changed the method name to lower case to avoid confusion! Methods used by other files are supposed to be upper case so that they are exported and able to be called after another file imports it,  and so helper methods are supposed to be lower case? 
Is this correct?
I will add more information about the TODO shortly .

Reply:
i can rename the l1,l2,l3, they are really just temp variables of me walking down to get to the spec data